design/README.md

```diff
@@ -0,0 +1,298 @@
+| `did`      | `String` | Primary Key | Yes    | Yes      |
+| `username` | `String` |             | Yes    | Yes      |
+| `email`    | `String` |             | No     | Yes      |
+
+## Sub Account
+
+A "Sub User" is an account where the relationship is mediated by a Full User. As such, Fission does not need to keep as much information about them.
+
+This relationship also has knock-on effects, such as billing the Sub Account's usage to the associated Full User's account (once billing is set up).
+
+A Sub Account has the following fields:
+
+| Field                     | Type     | Role        | Unique | Required |
+|---------------------------|----------|-------------|--------|----------|
+| `did`                     | `String` | Primary Key | Yes    | Yes      |
+| `manager_id`[^manager_id] | `String` | Foreign Key | No     | Yes      |
+
+[^manager_id]: Billing requires having this information stored directly in the database. However, control over a subaccount may be granted separately via UCAN. Having UCAN be the main mode of tracking authority is possible, but  This relationship may also be modelled with UCANs and capability store, but we have opted to not do that for reasons of efficient query and ease of enforcing invariants.
+
+# Capabilities
+
+To facilitate UCAN delegation when users are offline, the Fission server adapts some patterns from object capabilities. As new resources are delegated to an account's DID, the relevant UCAN is captured in a cache and indexed. This provides the ability to attach arbitrary capabilities to an account.
+
+In object oriented terms, this means that the account learns about new messages that it can call. From the perspective of the delegated root user, it means that new APIs (or RPC actions) are possibly available dynamically. The capabilities may or _may not_ be exposed to the delegated user ("account owner"), since these may be abstracted away by a server-managed resource ("object"). When delegatable, an endpoint should be available for the delegated user to download from the cache. Clearing this cache is not required except for after expiration or revocation.
+
+The UCAN that contains the capability is kept in binary storage (i.e. an object store). The capability is indexed for lookup performance:
+
+| Field         | Type       | Role        | Unique | Required |
+|---------------|------------|-------------|--------|----------|
+| `id`          | `Int`      | Primary Key | Yes    | Yes      |
+| `cid`         | `String`   |             | No     | Yes      |
+| `aud`         | `String`   | Foreign Key | No     | Yes      |
+| `uri`         | `String`   |             | No     | Yes      |
+| `ability`     | `String`   |             | No     | Yes      |
+| `caveat`      | `String`   |             | No     | No       |
+| `nbf`         | `DateTime` |             | No     | Yes      |
+| `exp`         | `DateTime` |             | No     | Yes      |
```

Comment:
I agree that something like this is useful in the general case, but I fail to see how it's needed for the User Stories.
Is this higher priority than Storage or Domains?

Reply:
Oh, yeah, definitely 💯 This needs that earlier disclaimer "this isn't available at time of writing". Will fix 👍

Reply:
https://starmap.site/roadmap/github.com/fission-codes/fission-server/issues/173#view=simple

cc @matheus23

Reply:
(Still a WIP, but coming along!)